examples/step-75/step-75.cc

```diff
@@ -0,0 +1,1575 @@
+            std::make_unique<AdaptationStrategy<dim>>(ParameterHandler(),
+                                                      dof_handler,
+                                                      triangulation);
+          break;
+        case AdaptationType::hpLegendre:
+          adaptation_strategy = std::make_unique<hpLegendreStrategy<dim>>(
+            ParameterHandler(), dof_handler, triangulation, fe_collection);
+          break;
+        case AdaptationType::hpFourier:
+          adaptation_strategy = std::make_unique<hpFourierStrategy<dim>>(
+            ParameterHandler(), dof_handler, triangulation, fe_collection);
+          break;
+        case AdaptationType::hpHistory:
+          adaptation_strategy =
+            std::make_unique<hpHistoryStrategy<dim>>(ParameterHandler(),
+                                                     dof_handler,
+                                                     triangulation);
+          break;
+        default:
+          AssertThrow(false, ExcNotImplemented());
+          break;
+      }
+  }
+
+
+
+  template <int dim>
+  void LaplaceProblem<dim>::create_coarse_grid()
+  {
+    TimerOutput::Scope t(computing_timer, "coarse grid");
+
+    std::vector<unsigned int> repetitions(dim, 2);
+    Point<dim>                bottom_left, top_right;
+    for (unsigned int d = 0; d < dim; ++d)
+      {
+        bottom_left[d] = -1.;
+        top_right[d]   = 1.;
+      }
+
+    std::vector<int> cells_to_remove(dim, 1);
+    cells_to_remove[0] = -1;
+
+    // TODO
+    // expand domain by 1 cell in z direction for 3d case
+
+    GridGenerator::subdivided_hyper_L(
+      triangulation, repetitions, bottom_left, top_right, cells_to_remove);
+  }
+
+
+
+  template <int dim>
+  void LaplaceProblem<dim>::setup_system()
+  {
+    TimerOutput::Scope t(computing_timer, "setup");
+
+    dof_handler.distribute_dofs(fe_collection);
+
+    locally_owned_dofs = dof_handler.locally_owned_dofs();
+    DoFTools::extract_locally_relevant_dofs(dof_handler, locally_relevant_dofs);
+
+    locally_relevant_solution.reinit(locally_owned_dofs,
+                                     locally_relevant_dofs,
+                                     mpi_communicator);
+    system_rhs.reinit(locally_owned_dofs, mpi_communicator);
+
+    constraints.clear();
+    constraints.reinit(locally_relevant_dofs);
+    DoFTools::make_hanging_node_constraints(dof_handler, constraints);
+    VectorTools::interpolate_boundary_values(dof_handler,
+                                             0,
+                                             Solution<dim>(),
+                                             constraints);
+
+#ifdef DEBUG
+    // We have not dealt with chains of constraints on ghost cells yet.
+    // Thus, we are content with verifying their consistency for now.
+    IndexSet locally_active_dofs;
+    DoFTools::extract_locally_active_dofs(dof_handler, locally_active_dofs);
+    AssertThrow(constraints.is_consistent_in_parallel(
```

Comment:
I think that's something we should put into a test, but maybe not a tutorial program